notion-distcopy/src/main/scala/com/ambiata/notion/distcopy/DistCopyInputFormat.scala

```diff
@@ -64,16 +65,17 @@ class DistCopyInputFormat extends InputFormat[NullWritable, Mapping] {
 }
 
 object DistCopyInputFormat {
-  val MappingKey = DistCache.Key("notion.dist-copy.sync.mappings")
+  val MappingKey = ThriftCache.Key("notion.dist-copy.sync.mappings")
+
+  private val local = new java.util.concurrent.ConcurrentHashMap[String, Workloads]
+
+  def take(context: MrContext): Option[Workloads] =
```

Comment:
This probably needs a comment, perhaps a link to the similar ivory structure, and just a justification that this is all in-memory on client side, so avoiding serialization saves memory and there is simply no way to pass it as an argument.

My bad on the Ivory one, it isn't justified well, and it looks odd, but I agree it is the best approach we have for this type of data.